modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptPathAttributeGenerator.cs

```diff
@@ -93,8 +95,19 @@ IEnumerable<ClassDeclarationSyntax> classDeclarations
                 if (attributes.Length != 0)
                     attributes.Append("\n");
 
+                string scriptPath = RelativeToDir(cds.SyntaxTree.FilePath, godotProjectDir);
+                if (!usedPaths.Add(scriptPath))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(
+                        Common.MultipleClassesInGodotScriptRule,
+                        cds.GetLocation(),
```

Comment:
For a better IDE experience, avoid raising the diagnostic on the entire class declaration syntax. Report on the symbol declaration instead.

```suggestion
                        cds.Identifier.GetLocation(),
```